unit_tests/test_name_resolving.py

```diff
@@ -0,0 +1,68 @@
+import struct
+
+from pytest_bdd import given, parsers, scenarios, then, when
+
+from boofuzz import *
+from boofuzz.mutation_context import MutationContext
+
+CONVERTERS = {
+    "block_name": str,
+}
+
+scenarios("test_name_resolving.feature", example_converters=CONVERTERS)
+
+
+@given("A flat scenario with block_name <block_name>")
+def a_flat_scenario(context, block_name):
+    global blocks
+    blocks.REQUESTS = {}
```

Comment:
Do we need `global blocks`? The tests seem to work fine without it.

Maybe we can reuse the fixture from `test_primitives.py` to clear the requests instead of calling `blocks.REQUESTS = {}` everytime.

```python
@pytest.fixture(autouse=True)
def clear_requests():
    yield
    blocks.REQUESTS = {}
    blocks.CURRENT = None
```

Reply:
Maybe there's a better way to import this function into the tests that need it, but I'll just add it to `test_name_resolving.py` for now.

Reply:
That'd be nice. Maybe we can add it to conftest.py? That would run it on all tests AFAIK.
But that's something for another time